### PR TITLE
Support for N of M signatures

### PR DESCRIPTION
**Pull request by fepitre**
https://github.com/QubesOS/qubes-issues/issues/7739

### CODE REVIEW

File: tests/test_scripts.py [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+        capture_output=True,
+    )
+    args = create_dummy_args(
+        component_repository=component_repository,
+        component_directory=temp_directory,
+        git_branch=git_branch,
+        fetch_versions_only=True,
+    )
+    get_and_verify_source(args)
+    assert (temp_directory / ".git/FETCH_HEAD").exists()
```
Comment (marmarek) [inline tests/test_scripts.py:244]: Check also if really version tag is on top. You might want to adjust initial clone (or add `git reset` just after) to have it few commits before version tag.

Similarly, a useful test would be with fresh clone with fetch_versions_only (pointing at branch where version tag is not on top but sometime earlier - and check if really that version was fetched).

File: tests/test_scripts.py [hunk 2]
```diff
@@ -0,0 +1,588 @@
...
+        component_directory=temp_directory,
+        fetch_versions_only=True,
+    )
+    get_and_verify_source(args)
+
+    # Check if we have a version tag on HEAD
+    vtag = subprocess.run(
+        [
+            "git",
+            "describe",
```
Comment (marmarek) [inline tests/test_scripts.py:295]: Describe (especially with `--abrev=0`) will find a tag, even if not on top. Here you want `git tag --points-at=HEAD v*` instead.

File: tests/test_scripts.py [hunk 3]
```diff
@@ -0,0 +1,588 @@
...
+        "--> Switching branch from main branch to new release4.1\n--> Merging...\n"
+        in capsys.readouterr().out
+    )
+    assert (temp_directory / ".git/FETCH_HEAD").exists()
+
+    # Check if we have a version tag on HEAD
+    vtag = subprocess.run(
+        [
+            "git",
+            "describe",
```
Comment (marmarek) [inline tests/test_scripts.py:326]: Here too.

File: tests/test_scripts.py [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+        git_branch="builderv2-tests-4",
+        fetch_versions_only=True,
+    )
+    get_and_verify_source(args)
+
+    # Check if we have a version tag on HEAD
+    vtag = subprocess.run(
+        [
+            "git",
+            "describe",
```
Comment (marmarek) [inline tests/test_scripts.py:578]: Here too.

**Review by marmarek — CHANGES_REQUESTED** [review]

File: tests/test_scripts.py (hunk 2 above)
Comment (fepitre) [reply tests/test_scripts.py]: Done.

**Review by marmarek — APPROVED** [review]

**Pull request merged**